### PR TITLE
Feature/administration page

### PR DESCRIPTION
**Pull request by JeanVittory**
# SUMMARY

Some endpoint from backend were adapted to frontend

## TO-DO

Keep migrating the backend services to the frontend

### CODE REVIEW

File: src/pages/administration/components/products/context/products.context.jsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ export const ProductsContext = ({ children }) => {
 
   const { products, productsError, productsIsLoading } = useProducts();
 
-  const categories = [...new Set(products?.map((product) => product.category))];
+  //const categories = [...new Set(products?.map((product) => product.category))];
```
Comment (ht1204) [inline src/pages/administration/components/products/context/products.context.jsx:23]: what is this commented code doing?

File: src/pages/administration/components/doctors/components/doctorsList/DoctorsList.doctors.administration.jsx [hunk 2]
```diff
@@ -1,51 +1,55 @@
...
-      });
-      if (status > 399) return errorMessage(errorMessage);
+      const ACCESS_TOKEN = localStorage.getItem(TOKEN);
+      if (!ACCESS_TOKEN) return navigate(PublicRoutes.LOGIN);
+      const { id } = doctor;
+      const payload = { id };
+      const isConfirmed = await confirmDeletion();
+      if (isConfirmed) {
+        await deleteDoctor(DELETE_DOCTOR, payload, ACCESS_TOKEN);
+        successMessage('Doctor deleted succesfully!');
```
Comment (ht1204) [inline src/pages/administration/components/doctors/components/doctorsList/DoctorsList.doctors.administration.jsx:48]: `successMessage('Doctor deleted successfully!');`

File: src/pages/administration/components/products/components/productForm/Form.products.administration.jsx [hunk 3]
```diff
@@ -80,27 +86,32 @@ export const Form = () => {
...
+      if (!isToken) return navigate(PublicRoutes.LOGIN);
+      const { image } = newProduct;
+      if (typeof image === 'string') {
+        const { image, category, newCategory, ...toUpdate } = newProduct;
+        const form = new FormData();
+        for (const key in toUpdate) {
+          form.append(key, toUpdate[key]);
+        }
+        await updateProducts(UPDATE_PRODUCTS, form, isToken);
+        successMessage('Product updated succesfully!');
```
Comment (ht1204) [inline src/pages/administration/components/products/components/productForm/Form.products.administration.jsx:101]: `successMessage('Product updated successfully!');`

File: src/pages/login/Login.page.jsx [hunk 4]
```diff
@@ -38,55 +37,52 @@ export const Login = () => {
...
               type="text"
               ref={inputName}
-              placeholder="Username"
-              name="username"
-              id="username"
+              placeholder="Email"
+              name="email"
+              id="email"
               minLength="3"
               pattern="[A-Za-z\s]{3,}"
```
Comment (ht1204) [inline src/pages/login/Login.page.jsx:81]: this regex would be better in a constant

File: src/pages/login/Login.page.jsx [hunk 5]
```diff
@@ -114,14 +110,7 @@ export const Login = () => {
...
               <label htmlFor="done">Remember me</label>
-              <input
-                id="done"
-                name="remberMe"
-                type="checkbox"
-                className="remember__button"
-                onChange={handleChange}
-                checked={remember}
-              />
+              <input id="done" name="remberMe" type="checkbox" className="remember__button" />
```
Comment (ht1204) [inline src/pages/login/Login.page.jsx:113]: `rememberMe`

File: src/pages/login/service/auth.service.js [hunk 6]
```diff
@@ -3,38 +3,38 @@ import { authAdapter } from '../adapters';
 
 export const authService = async (URL, user) => {
   try {
-    const response = await axios.post(URL, user);
-    const isValid = validateUser(user, response.data);
-    if (isValid instanceof Error) throw isValid;
-    return authAdapter(isValid);
+    const { data } = await axios.post(URL, user);
+    //const isValid = validateUser(user, response.data);
+    //if (isValid instanceof Error) throw isValid;
```
Comment (ht1204) [inline src/pages/login/service/auth.service.js:8]: delete this commented code, please.

File: src/pages/register/Register.page.jsx [hunk 7]
```diff
@@ -1,74 +1,79 @@
...
+            gender: '',
+            birthdate: Date.now(),
+            nationality: '',
+            password: '',
+            blood_type: '',
+            repeatPassword: '',
+            termsAndConditions: false,
+          });
+          navigate(PublicRoutes.LOGIN);
+          return successMessage('Your register was success!');
```
Comment (ht1204) [inline src/pages/register/Register.page.jsx:62]: `return successMessage('Your register was succeeded!');`

File: src/pages/register/Register.page.jsx [hunk 8]
```diff
@@ -1,74 +1,79 @@
...
+        } else {
+          errorMessage('You must agree to terms and conditions');
```
Comment (ht1204) [inline src/pages/register/Register.page.jsx:64]: `You must accept the terms and conditions'`

File: src/pages/administration/components/products/context/products.context.jsx (hunk 1 above)
Comment (JeanVittory) [reply src/pages/administration/components/products/context/products.context.jsx:23]: Done!

File: src/pages/login/Login.page.jsx (hunk 4 above)
Comment (JeanVittory) [reply src/pages/login/Login.page.jsx:81]: Done

File: src/pages/register/Register.page.jsx (hunk 7 above)
Comment (JeanVittory) [reply src/pages/register/Register.page.jsx:62]: done!

File: src/pages/login/service/auth.service.js [hunk 9]
```diff
@@ -3,38 +3,38 @@ import { authAdapter } from '../adapters';
...
+//     };
+//   } else if (userExist.username === 'Jean Vittory') {
+//     return {
+//       ACCESS_TOKEN: '123',
+//     };
+//   } else {
+//     return {
+//       ACCESS_TOKEN: '9876543210',
+//     };
+//   }
```
[the remark this answers is not in the document]
Comment (JeanVittory) [reply src/pages/login/service/auth.service.js:39]: done!

File: src/pages/administration/components/doctors/components/doctorsList/DoctorsList.doctors.administration.jsx (hunk 2 above)
Comment (JeanVittory) [reply src/pages/administration/components/doctors/components/doctorsList/DoctorsList.doctors.administration.jsx:48]: done

**Review by ht1204 — APPROVED** [review]

**Pull request merged**